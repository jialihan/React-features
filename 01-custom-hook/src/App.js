import React from "react";
import logo from "./logo.svg";
import "./App.css";
import ColorTool from "./components/ColorTool/ColorTool";

function App() {
  return (
    <div className="App">
      <ColorTool />
    </div>
  );
}

export default App;
