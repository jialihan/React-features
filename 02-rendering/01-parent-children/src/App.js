
import './App.css';
import { Child } from './components/Child/Child';
import { Container } from './components/Container/Container';

function App() {
  return (
    <div className="App">
      <Container>
        <Child />
      </Container>
    </div>
  );
}

export default App;
