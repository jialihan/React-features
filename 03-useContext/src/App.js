import logo from './logo.svg';
import './App.css';
import { Authcomponent } from './component/AuthComponent';
import { AuthComponent_consumer } from './component/AuthComponent_consumer';

function App() {
  return (
    <div className="App">
      <h1>React Context Example</h1>
      <Authcomponent />
      <AuthComponent_consumer />
    </div>
  );
}

export default App;
