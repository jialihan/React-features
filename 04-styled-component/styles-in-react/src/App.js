import logo from './logo.svg';
import './App.css';
import InlineCSSComponent from './component/inlineComponent';
import Normalcsscomponent from './component/NormalCSSComponent';
import Styledcsscomponent from './component/StyledCSSComponent';

function App() {

  return (
    <div>
      <InlineCSSComponent />
      <Normalcsscomponent />
      <Styledcsscomponent />
    </div>
  );
}

export default App;
