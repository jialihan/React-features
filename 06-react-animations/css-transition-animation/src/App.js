import './App.css';
import ErrorMain from './components/CSSTransition/errorComponent';
import Main from './components/CSSTransition/index'
function App() {
  return (
    <div className="App">
      {/* <Main /> */}
      <ErrorMain />
    </div>
  );
}

export default App;
