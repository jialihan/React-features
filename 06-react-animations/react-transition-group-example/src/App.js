import './App.css';
import AnimationModal from './components/AnimationModal/AnimationModal';
import TransitionComponent from './components/SimpleTransitionComponentExample/TransitionComponent.jsx'
import List from './components/List/List'
import TransitionGroupList from './components/List/TransitionGroupList';
function App() {
  return (
    <div className="App">
      <TransitionComponent />
      <AnimationModal />
      <h1>Animating Lists</h1>
      <TransitionGroupList />
    </div>
  );
}

export default App;
