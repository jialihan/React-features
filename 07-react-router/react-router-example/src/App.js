import "./App.css";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import Home from "./components/Home.jsx";
import User from "./components/User.jsx";
import Admin from "./components/Admin.jsx";
import Nav from "./components/Nav.jsx";
import Manage from "./components/Manage.jsx";
import RedirectHandler from "./components/RedirectHandler.jsx";
import { AuthContextProvider } from "./context/AuthContext.jsx";
import ProtectedRoute from "./components/ProtectedRoute.jsx";
import RoleSelector from "./components/RoleSelector.jsx";

function App() {
  return (
    <AuthContextProvider>
      <BrowserRouter>
        <RoleSelector />
        <Nav />
        <Routes>
          <Route
            path="/admin"
            element={
              <ProtectedRoute>
                <Admin />
              </ProtectedRoute>
            }
          />
          <Route
            path="/user"
            element={
              <ProtectedRoute>
                <User />
              </ProtectedRoute>
            }
          />
          <Route
            path="/manage"
            element={
              <ProtectedRoute>
                <Manage />
              </ProtectedRoute>
            }
          />
          <Route
            path="/"
            element={
              <RedirectHandler>
                <Home />
              </RedirectHandler>
            }
          />
        </Routes>
      </BrowserRouter>
    </AuthContextProvider>
  );
}

export default App;
