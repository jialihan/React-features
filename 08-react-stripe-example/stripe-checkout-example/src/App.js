import logo from './logo.svg';
import './App.css';
import StripeCheckoutButton from './components/StripeCheckoutButton';

function App() {
  const price = 99.99;
  return (
    <div className="App">
      <h1> your total is ${price}</h1>
      <StripeCheckoutButton price={price} />
      <p className="warning-message">
        *Please use test credit card for payments*
        <br />
        4242424242424242 |	Visa |	Any 3 digits|	Any future date
      </p>
    </div>
  );
}

export default App;
