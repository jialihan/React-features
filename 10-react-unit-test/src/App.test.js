// test file
import { shallow } from 'enzyme';
import App from './App';


// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
let wrapper;

describe("Component should render", () => {
  it("should match snapshot", () => {
    wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
