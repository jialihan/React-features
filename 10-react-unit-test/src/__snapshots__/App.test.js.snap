// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should render should match snapshot 1`] = `ShallowWrapper {}`;
